lang/syn/src/lib.rs

```diff
@@ -662,11 +662,14 @@ pub enum ConstraintToken {
     Address(Context<ConstraintAddress>),
     TokenMint(Context<ConstraintTokenMint>),
     TokenAuthority(Context<ConstraintTokenAuthority>),
+    TokenTokenProgram(Context<ConstraintTokenProgram>),
     AssociatedTokenMint(Context<ConstraintTokenMint>),
     AssociatedTokenAuthority(Context<ConstraintTokenAuthority>),
+    AssociatedTokenProgram(Context<ConstraintTokenProgram>),
```

Comment:
Same here, consistency with `TokenTokenProgram` above and make it more clear.

```suggestion
    AssociatedTokenTokenProgram(Context<ConstraintTokenProgram>),
```

Reply:
Sounds good